### DISCUSSION TITLE
How Can I Change the Primary Color?

### DISCUSSION THREAD

**Discussion by k-wakamatsu-tms** [chakra-ui/chakra-ui#9291] [Q&A] [answered]
How can I change the overall primary color from the default, similar to the playground?

I’ve made the following modifications, but the color does not change:

```ts
// theme.ts
import {
  createSystem,
  defaultBaseConfig,
  defaultConfig,
  defineConfig,
  mergeConfigs,
} from "@chakra-ui/react";

const customConfig = defineConfig({
  theme: {
    tokens: {
      colors: {
        primary: { value: "#0FEE0F" },
        secondary: { value: "#EE0F0F" },
      },
    },
  },
});
const config = mergeConfigs(defaultConfig, customConfig);
export const system = createSystem(config);
```

```tsx
// provider.tsx
"use client";

import { ChakraProvider, defaultSystem } from "@chakra-ui/react";
import { ColorModeProvider, type ColorModeProviderProps } from "./color-mode";
import { system } from "theme";

export function Provider(props: ColorModeProviderProps) {
  return (
    <ChakraProvider value={system}>
      <ColorModeProvider {...props} />
    </ChakraProvider>
  );
}
```

**Comment by segunadebayo** [comment, marked as answer]
Here you go https://github.com/chakra-ui/chakra-ui/blob/main/apps/www/content/guides/theming-change-default-color-palette.mdx

**Reply by veloware** [reply]
I get TS error with the above example - 

<img width="871" alt="image" src="https://github.com/user-attachments/assets/c45be113-12ca-41a8-a983-f526a5ac20c2">


but this works - 

<img width="628" alt="image" src="https://github.com/user-attachments/assets/b2540a4a-62fe-4847-bb6b-a1d27ce0bda1">

i.e. globalCss rather than globalStyles, and moved to the top level of the config, and not within `theme`

**Reply by k-wakamatsu-tms** [reply]
Thank you.

I also encountered a TypeScript error with the above code, so I resolved it using `globalCss`.

Another solution was to wrap the entire app with `Theme`, which worked as well. However, is this approach not recommended?

```tsx
export default function App() {
  return (
    <Provider>
      <Theme colorPalette={"green"}>
        <Outlet />
      </Theme>
    </Provider>
  );
}
```
